app/views/users/index.html.erb

```diff
@@ -6,6 +6,25 @@
         <span class="profile-link">
           <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
         </span>
+
+       
+        <% if current_user.friend_invites(user) %>
+            <button  class = "pending"> Pending Invitation </button>
+          <% elsif current_user.receive_invitation(user)%>
+          <button  class = "accept"> <%= link_to 'Accept', accept_invite_path(user_id: user.id) %> </button>
+        <button class = "decline"> <%= link_to 'Decline', reject_invite_path(user_id: user.id), method: :delete, class:"btn btn-danger" %></button>
+          <% elsif !current_user.friend?(user) && current_user.id != user.id %>
+            <button class = "add_friend"><%= link_to 'Add Friend', send_request_path(user_id: user.id) , class:"btn btn-success" %></button>
+            <% elsif current_user.friend?(user) %>
+            <button class = "add_friend">Added</button>
+          <% end %>
```

Comment:
**Optional**. It is not a good practice to have logic in the views. You can move the if/ else statements to a view helper instead.

Regarding the if/else statements. Some of the logic can be implemented from the controller(as a variable) or from the Model (that's the best option) as a method, instead of here.